### PR TITLE
fix: [CDS-45827]: variables api refetched on service inline edit inside pipeline studio

### PR DESCRIPTION
**Pull request by bhavya-sinha**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [x] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [x] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by bhavya-sinha** [comment]
retrigger jest

**Comment by bhavya-sinha** [comment]
retrigger cypress-pipeline

**Review by udayvunnam — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx [hunk 1]
```diff
@@ -316,6 +319,7 @@ export default function DeployServiceEntityWidget({
           )
         )
       }
+      await refetchPipelineVariable?.()
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx:322]: We don't need await here

Comment (bhavya-sinha) [reply src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx:322]: Will correct in next PR

**Comment by bhavya-sinha** [comment]
retrigger cypress-rest

**Review by satyam-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**